tdrs-frontend/src/components/RequestAccessForm/RequestAccessForm.jsx

```diff
@@ -0,0 +1,358 @@
+        typeof fieldValue === 'string' &&
+        fieldValue.trim() === '') ||
+      (fieldValue instanceof Set && fieldValue.size === 0)
+    ) {
+      return `${field} is required`
+    }
+    return null
+  }
+
+  const setJurisdictionType = (val) => {
+    setStt('')
+    setJurisdictionTypeInputValue(val)
+  }
+
+  const setStt = (sttName) => {
+    setProfileInfo((currentState) => ({
+      ...currentState,
+      stt: sttName,
+    }))
+  }
+
+  const handleChange = ({ name, value }) => {
+    setProfileInfo({ ...profileInfo, [name]: value })
+  }
+
+  const handleBlur = (evt) => {
+    const { name, value } = evt.target
+
+    const { [name]: removedError, ...rest } = errors
+
+    const error = validation(name, value)
+
+    setErrors({
+      ...rest,
+      ...(error && { [name]: touched[name] && error }),
+    })
+  }
+
+  const handleSubmit = (evt) => {
+    evt.preventDefault()
+
+    // validate the form
+    const formValidation = Object.keys(profileInfo).reduce(
+      (acc, key) => {
+        const newError = validation(key, profileInfo[key])
+        const newTouched = { [key]: true }
+        return {
+          errors: {
+            ...acc.errors,
+            ...(newError && { [key]: newError }),
+          },
+          touched: {
+            ...acc.touched,
+            ...newTouched,
+          },
+        }
+      },
+      {
+        errors: { ...errors },
+        touched: { ...touched },
+      }
+    )
+    setErrors(formValidation.errors)
+    setTouched(formValidation.touched)
+
+    if (!Object.values(formValidation.errors).length) {
+      return dispatch(
+        requestAccess({
+          ...profileInfo,
+          stt: sttList.find((stt) => stt.name === profileInfo.stt),
+        })
+      )
+    }
+    return setTimeout(() => errorRef.current.focus(), 0)
+  }
+
+  return (
+    <div className="margin-top-5 margin-bottom-5">
+      <p className="margin-top-1 margin-bottom-4">
+        Please enter your information to request access from an OFA
+        administrator
+      </p>
+      <p>Fields marked with an asterisk (*) are required.</p>
+      <form className="usa-form" onSubmit={handleSubmit}>
+        <div
+          className={`usa-error-message ${
+            displayingError ? 'display-block' : 'display-none'
+          }`}
+          ref={errorRef}
+          tabIndex="-1"
+          role="alert"
+        >
+          There are {Object.keys(errors).length} errors in this form
+        </div>
+        <FormGroup
+          error={errors.firstName}
+          name="firstName"
+          label="First Name"
+          inputValue={profileInfo.firstName}
+          handleChange={handleChange}
+          handleBlur={handleBlur}
+        />
+        <FormGroup
+          error={errors.lastName}
+          name="lastName"
+          label="Last Name"
+          inputValue={profileInfo.lastName}
+          handleChange={handleChange}
+          handleBlur={handleBlur}
+        />
+        {isAMSUser && (
```

Comment:
if you wanted to take the component-ization a little further, the three longer conditional renders look like good candidates for dumb/display components that utilize props passed in by this component. Otherwise your state handling and functions at the top level look really good!